pylearn2/datasets/cache.py

```diff
@@ -116,7 +116,14 @@ def cache_file(self, filename):
 
         # Create the folder structure to receive the remote file
         local_folder = os.path.split(local_name)[0]
-        self.safe_mkdir(local_folder)
+        try:
+            self.safe_mkdir(local_folder)
+        except Exception, e:
```

Comment:
This syntax does not work for Python 3, I think it should be
    except Exception as e: